lib/neovim-plugin.nix

```diff
@@ -113,6 +113,30 @@
                 '';
                 internal = true;
               };
+
+              lazyLoad = lib.mkOption {
+                default = {
+                  enable = false;
+                };
+                description = ''
+                  Lazy load configuration settings.
+                '';
+                type = lib.types.submodule {
+                  options = {
+                    enable = lib.mkOption {
```

Comment:
I'm wondering, if it is likely we will have a different "enable" in the future, should this be named something else?

Having it named something like `dontCallSetup` may be more explicit and would also allow us to more easily deprecate this option if we need to in the future?